pkg/volume/git_repo/git_repo.go

```diff
@@ -138,24 +141,39 @@ func (b *gitRepoVolumeBuilder) SetUpAt(dir string) error {
 		return err
 	}
 
-	if output, err := b.execCommand("git", []string{"clone", b.source}, dir); err != nil {
-		return fmt.Errorf("failed to exec 'git clone %s': %s: %v", b.source, output, err)
+	args := []string{"clone", b.source}
+
+	if b.target != "" {
+		args = append(args, b.target)
+	}
+	if output, err := b.execCommand("git", args, dir); err != nil {
+		return fmt.Errorf("failed to exec 'git %s': %s: %v",
+			strings.Join(append(args[:1], args[1:]...), " "), output, err)
 	}
 
 	files, err := ioutil.ReadDir(dir)
 	if err != nil {
 		return err
 	}
-	if len(files) != 1 {
-		return fmt.Errorf("unexpected directory contents: %v", files)
-	}
+
 	if len(b.revision) == 0 {
```

Comment:
` len(b.revision) == 0` and `b.target != ""` are the same (in effect) but different implementations.  They should be consistent.  I've mostly seen (and used) empty string "" comparison instead of string length check.